lib/developer_portal/lib/liquid/url_helper_hacks.rb

```diff
@@ -22,8 +22,9 @@ def current_page?(options)
     private
 
     # Semi-hacky way to access controller from within liquid's filters and tags.
-    def controller
-      @context.registers[:controller]
-    end
+    # TODO: Ruby 3 TEMPORARY HACK! Figure this out for liquid
+    # def controller
+    #   @context.registers[:controller]
+    # end
```

Comment:
What is the consequence of this method being removed?

Reply:
Well, this is tricky. I just commented it out **temporarily** in order to be able to load the admin portal and find other issues :upside_down_face: 

This needs a proper fix.

Basically, what the issue is is that I think this is supposed to be only loaded on the developer portal, where we have Liquid. But it was also loading in the admin portal, so each time any admin page was loaded, there was this error:
```
ActionView::Template::Error (undefined method `registers' for nil:NilClass

      @context.registers[:controller]
              ^^^^^^^^^^):
    1: - content_for :stylesheets do
    2:   = stylesheet_link_tag 'vendor/c3'
    3: 
    4: - content_for :javascripts do
    5:   = javascript_packs_with_chunks_tag 'dashboard'
```
Because we don't have `@context` here.

We're loading that helper here:
https://github.com/3scale/porta/blob/cf17e8c8bd51902c1fa56d8fce31522b36541ee1/lib/developer_portal/config/initializers/liquid.rb#L9-L11

and I couldn't figure out yet how to load it only for the developer portal.

And in general, I don't quite understand how Liquid works, where `@context` is populated etc. So I left it for later.
If you have any ideas - please let me know!

Reply:
Maybe full backtrace will help identify where is it called from.

Reply:
I ended up removing the whole thing :grimacing: 
Some comments about it here - https://github.com/3scale/porta/pull/3847#discussion_r1686455673